src/mem/epoch/mod.rs

```diff
@@ -429,7 +429,14 @@ impl Guard {
     /// Assert that the value is no longer reachable from a lock-free data
     /// structure and should be collected when sufficient epochs have passed.
     pub unsafe fn unlinked<T>(&self, val: Shared<T>) {
-        local::with_participant(|p| p.reclaim(val.as_raw()))
+        local::with_participant(|p| p.reclaim(val.as_raw(), false))
+    }
+
+    /// Assert that the value is no longer reachable from a lock-free data
+    /// structure and should be collected and destroyed
+    /// when sufficient epochs have passed.
+    pub unsafe fn unlinked_drop<T: Drop>(&self, val: Shared<T>) {
```

Comment:
Like @Vtec234 said, this function needs `T: Drop + Send + 'static`.

* `Send` because some other thread might pick up the job of destructing `val`.
* `'static` because we don't know how soon `val` will be destructed (potentially never).